src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -0,0 +1,303 @@
+    error: profileError,
+  } = useQuery({
+    queryKey: [queryKeys.profiles],
+    queryFn: () => fetchProfiles(project),
+  });
+
+  if (profileError) {
+    notify.failure("Loading profiles failed", profileError);
+  }
+
+  const inheritedProxies = getInheritedProxies(formik.values, profiles);
+
+  const existingDeviceNames: string[] = [];
+  existingDeviceNames.push(...inheritedProxies.map((item) => item.key));
+  existingDeviceNames.push(...formik.values.devices.map((item) => item.name));
+
+  const addProxyCard = (card: ProxyCard) => {
+    const copy = [...formik.values.devices];
+    copy.push({
+      type: "proxy",
+      bind: card.bind,
+      nat: card.nat,
+      proxy_protocol: card.proxy_protocol,
+      mode: card.listen?.mode,
+      listen: `${card.listen?.type}:${card.listen?.address}:${card.listen?.port}`,
+      connect: `${card.listen?.type}:${card.listen?.address}:${card.listen?.port}`,
+      name: deduplicateName("proxy", 1, existingDeviceNames),
+    });
+
+    void formik.setFieldValue("devices", copy);
+  };
+
+  const hasCustomProxy = formik.values.devices.some(
+    (item) => item.type === "proxy",
+  );
+
+  const inheritedRows: MainTableRow[] = [];
+  inheritedProxies.forEach((item) => {
+    const noneDeviceId = findNoneDeviceIndex(item.key, formik);
+    const isNoneDevice = noneDeviceId !== -1;
+
+    inheritedRows.push(
+      getConfigurationRowBase({
+        className: "no-border-top override-with-form",
+        configuration: (
+          <div
+            className={classnames("device-name", {
+              "u-text--muted": isNoneDevice,
+            })}
+          >
+            <b>{item.key}</b>
+          </div>
+        ),
+        inherited: (
+          <div className="p-text--small u-text--muted u-no-margin--bottom">
+            From: {item.source}
+          </div>
+        ),
+        override: isNoneDevice ? (
+          <Button
+            appearance="base"
+            type="button"
+            title="Reattach volume"
+            onClick={() => {
+              ensureEditMode(formik);
+              removeDevice(noneDeviceId, formik);
+            }}
+            className="has-icon u-no-margin--bottom"
+          >
+            <Icon name="connected"></Icon>
+            <span>Reattach</span>
+          </Button>
+        ) : (
+          <Button
+            appearance="base"
+            type="button"
+            onClick={() => {
+              ensureEditMode(formik);
+              addNoneDevice(item.key, formik);
+            }}
+            className="has-icon u-no-margin--bottom"
+            dense
+          >
+            <Icon name="disconnect"></Icon>
+            <span>Detach</span>
+          </Button>
+        ),
+      }),
+    );
+
+    Object.keys(item.proxy).forEach((key) => {
+      if (key === "name" || key === "type") {
+        return null;
+      }
+
+      inheritedRows.push(
+        getInheritedDeviceRow({
+          label: deviceKeyToLabel(key),
```

Comment:
Here we might encounter any field in the proxy config (see docs for a [full list](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_proxy/#device-options)). So we should mention all possible keys in the `deviceKeyToLabel` function, also `uid` and alike, even though we don't support them otherwise in the ui.
For testing you could add them via the YAML editor to a proxy device.

Reply:
Is that not the current behaviour?

Reply:
You are right, uid and gid is already in the deviceKeyToLabel function, i think the security.uid and security.gid are missing and there might be others?

Reply:
I can add those. They should be in the ProxyDevice type/interface but I was not sure whether to add them.